.eslintrc.js

```diff
@@ -40,6 +40,8 @@ module.exports = {
     '@typescript-eslint'
   ],
   rules: {
+    "no-use-before-define": "off", //recommended by TS lint
+    "@typescript-eslint/no-use-before-define": ["error"], //recommended by TS lint
```

Comment:
some components began throwing the default alert due to not understanding typescript fully - the TS one fully understands how to trace it.